File: test/data/cpp/test05/StateState1.cpp
#include "StateState1.h"
StateState1::StateState1(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState1::~StateState1()
{
}
BaseState* StateState1::TransitEvent2()
{
    _controllee->DoAction2();
    _stateController->InstanceOfState2->Setup();
    return _stateController->InstanceOfState2;
}
BaseState* StateState1::TransitEvent21()
{
    _controllee->DoAction21();
    _stateController->InstanceOfState21->Setup();
    return _stateController->InstanceOfState21;
}
BaseState* StateState1::TransitEvent31()
{
    _controllee->DoAction31();
    _stateController->InstanceOfState31->Setup();
    return _stateController->InstanceOfState31;
}
const char* StateState1::GetStateName()
{
    return "State1";
}
BaseState* StateState1::GetParent()
{
    return 0;
}

File: test/data/cs/test05/StateState1.cs
public class StateState1 : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateState1(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitEvent2()
    {
        _controllee.DoAction2();
        _stateController.InstanceOfState2.Setup(false, false);
        return _stateController.InstanceOfState2;
    }
    public override BaseState? TransitEvent21()
    {
        _controllee.DoAction21();
        _stateController.InstanceOfState21.Setup(false, false);
        return _stateController.InstanceOfState21;
    }
    public override BaseState? TransitEvent31()
    {
        _controllee.DoAction31();
        _stateController.InstanceOfState31.Setup(false, false);
        return _stateController.InstanceOfState31;
    }
    public override string GetStateName()
    {
        return "State1";
    }
    public override BaseState? GetParent()
    {
        return null;
    }
    public override int GetStateID()
    {
        return 0;
    }
}